server/src/main/scala/org/http4s/server/middleware/authentication/NonceKeeperF.scala

```diff
@@ -42,14 +43,15 @@ private[authentication] object NonceKeeperF {
     currentMillis <- Clock[F].monotonic(MILLISECONDS)
     lastCleanupMillis <- Ref[F].of(currentMillis)
     nonces = new LinkedHashMap[String, NonceF[F]]
+    random <- Random.javaSecuritySecureRandom[F](blocker)
```

Comment:
This is as high up as we can pull `Random` without making it public API.  In Cats-Effect 3, we can consider surfacing it all the way to the top.